src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/ApplicationTest.kt

```diff
@@ -3068,20 +3073,23 @@ class ApplicationTest : IntegrationTestBase() {
           val updatedAssessment = approvedPremisesAssessmentRepository.findByIdOrNull(assessment.id)!!
           assertThat(updatedAssessment.isWithdrawn).isTrue
 
-          val updatedBooking = bookingRepository.findByIdOrNull(booking1.id)!!
-          val cancellation = updatedBooking.cancellation
-          assertThat(cancellation).isNotNull
-          assertThat(cancellation!!.reason.name).isEqualTo("The probation practitioner requested it")
-
-          val updatedBookingWithArrival = bookingRepository.findByIdOrNull(bookingWithArrival.id)!!
-          assertThat(updatedBookingWithArrival.cancellation).isNull()
-
           val updatedPlacementApplication = placementApplicationRepository.findByIdOrNull(placementApplication.id)!!
           assertThat(updatedPlacementApplication.decision).isEqualTo(PlacementApplicationDecision.WITHDRAWN_BY_PP)
 
-          val updatedPlacementRequest = placementRequestRepository.findByIdOrNull(placementRequest.id)!!
-          assertThat(updatedPlacementRequest.isWithdrawn).isEqualTo(true)
-          assertThat(updatedPlacementRequest.withdrawalReason).isEqualTo(PlacementRequestWithdrawalReason.WITHDRAWN_BY_PP)
+          val updatedPlacementRequest1 = placementRequestRepository.findByIdOrNull(placementRequest1.id)!!
+          assertThat(updatedPlacementRequest1.isWithdrawn).isEqualTo(true)
+          assertThat(updatedPlacementRequest1.withdrawalReason).isEqualTo(PlacementRequestWithdrawalReason.WITHDRAWN_BY_PP)
+
+          val updatedPlacementRequest2 = placementRequestRepository.findByIdOrNull(placementRequest2.id)!!
+          assertThat(updatedPlacementRequest2.isWithdrawn).isEqualTo(true)
+          assertThat(updatedPlacementRequest2.withdrawalReason).isEqualTo(PlacementRequestWithdrawalReason.WITHDRAWN_BY_PP)
+
+          val updatedBooking1 = bookingRepository.findByIdOrNull(booking1NoArrival.id)!!
+          assertThat(updatedBooking1.isCancelled).isTrue()
+          assertThat(updatedBooking1.cancellation!!.reason.name).isEqualTo("The probation practitioner requested it")
+
+          val updatedBooking2WithArrival = bookingRepository.findByIdOrNull(booking2HasArrival.id)!!
+          assertThat(updatedBooking2WithArrival.isCancelled).isFalse()
```

Comment:
do you think it's worth showing here that an `updatedBooking1NoArrival ` has not been cancelled/withdrawn?

Reply:
Yes, absolutely agree. I actually have a PR ready to go with lots of tests for cascading that checks this and much more, so I think that'll cover it.